pkg/api/types.go

```diff
@@ -438,6 +440,21 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// CinderPersistentDisk represents a Persistent Disk resource in Openstack.
+// A Cinder volume must exist and be formatted before mounting to a container.
+// The disk must also be in the same region as the kubelet.
+type CinderPersistentDiskVolumeSource struct {
```

Comment:
Maybe just "CinderVolumeSource"?  It will be present in both the VolumeSource and PersistentVolumeSource structs.